quantum/pointing_device/pointing_device_modes.h

```diff
@@ -0,0 +1,176 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <string.h>
+#include "quantum.h"
+#include "pointing_device.h"
+
+/* sanity check */
+#if !defined(POINTING_DEVICE_MODES_ENABLE) || !defined(POINTING_DEVICE_ENABLE)
+#    error "POINTING_DEVICE_MODES_ENABLE or POINTING_DEVICE_ENABLE not defined! check settings!"
+#endif
+
+/* default settings */
+#ifndef POINTING_TAP_DELAY
+#    define POINTING_TAP_DELAY 0
+#endif
+#ifndef POINTING_MODE_DEFAULT
+#    define POINTING_MODE_DEFAULT PM_NONE
+#endif
+#if defined(SPLIT_POINTING_ENABLE) && defined(POINTING_DEVICE_COMBINED)
+#    if defined(POINTING_MODES_LEFT)
+#        define POINTING_MODES_LEFT_DEFAULT true
+#    else
+#        define POINTING_MODES_LEFT_DEFAULT false
+#    endif
+#endif
+
+/* default divisors */
+#ifndef POINTING_DEFAULT_DIVISOR
+#    define POINTING_DEFAULT_DIVISOR 64
+#endif
+#ifndef POINTING_HISTORY_DIVISOR
+#    define POINTING_HISTORY_DIVISOR 64
+#endif
+#ifndef POINTING_VOLUME_DIVISOR
+#    define POINTING_VOLUME_DIVISOR 64
+#endif
+#ifndef POINTING_CARET_DIVISOR
+#    define POINTING_CARET_DIVISOR 32
+#endif
+#ifndef POINTING_CARET_DIVISOR_H
+#    define POINTING_CARET_DIVISOR_H POINTING_CARET_DIVISOR
+#endif
+#ifndef POINTING_CARET_DIVISOR_V
+#    define POINTING_CARET_DIVISOR_V POINTING_CARET_DIVISOR
+#endif
+#ifndef POINTING_PRECISION_DIVISOR
+#    define POINTING_PRECISION_DIVISOR 2
+#endif
+#ifndef POINTING_DRAG_DIVISOR
+#    define POINTING_DRAG_DIVISOR 4
+#endif
+
+/* error checking */
+#if (POINTING_DEFAULT_DIVISOR == 0)
+#    error "ERROR: DEFAULT DIVISOR is set to zero! set to non zero value"
+#endif
+/* check valid keycode range */
+#if (!((POINTING_MODE_COUNT - 1) & (~POINTING_MODE_COUNT)))
+#    error "!!ERROR: number of keycodes for QK_POINTING_MODE incorrect, must be a power of 2!!"
+#endif
+
+/* keycode setup */
+#define POINTING_MODE_MO QK_POINTING_MODE
+#define POINTING_MODE_MO_MAX (POINTING_MODE_MO + POINTING_MODE_COUNT - 1)
+#define POINTING_MODE_TG (POINTING_MODE_MO_MAX + 1)
+#define POINTING_MODE_TG_MAX QK_POINTING_MODE_MAX
+
+/* Default Pointing device pointing modes */
+enum pointing_device_mode_list {
+    // Built in pointing modes OVERWRITABLE
+    PM_HISTORY = POINTING_MODE_COUNT - 6,
+    PM_VOLUME,
+    PM_CARET,
+    PM_PRECISION,
+    PM_DRAG,
+    // null pointing mode NOT OVERWRITABLE
+    PM_NONE = POINTING_MODE_COUNT - 1,
+    // safe range for custom modes
+    PM_SAFE_RANGE
+};
+
+/* pointing mode aliases */
+#define PM_PRE PM_PRECISION
+#define PM_HST PM_HISTORY
+#define PM_VOL PM_VOLUME
+#define PM_CRT PM_CARET
+#define PM_DRG PM_DRAG
```

Comment:
One issue here, what happens if you compile with these in a keymap and with the scroll mode define? 

Eg, we generally want to have it compiling, no matter what.

Reply:
Would it be better to do it that way rather than just throw the enum and aliases back into `quantum_keycodes.h`?

Reply:
I suppose the stub approach could be a bit cleaner and prevents the cluttering of `quantum_keycodes.h` so I'll make this change let me know which version you prefer once you get a chance to review again.